src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNullVisitor.java

```diff
@@ -17,63 +17,96 @@
 
 import lombok.EqualsAndHashCode;
 import lombok.Value;
-import org.jspecify.annotations.Nullable;
 import org.openrewrite.Tree;
 import org.openrewrite.java.JavaVisitor;
 import org.openrewrite.java.MethodMatcher;
 import org.openrewrite.java.style.EqualsAvoidsNullStyle;
-import org.openrewrite.java.tree.*;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Space;
 import org.openrewrite.marker.Markers;
 
 import static java.util.Collections.singletonList;
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+import static org.openrewrite.java.tree.JLeftPadded.build;
+import static org.openrewrite.java.tree.Space.SINGLE_SPACE;
 
 @Value
 @EqualsAndHashCode(callSuper = false)
 public class EqualsAvoidsNullVisitor<P> extends JavaVisitor<P> {
-    private static final MethodMatcher STRING_EQUALS = new MethodMatcher("String equals(java.lang.Object)");
-    private static final MethodMatcher STRING_EQUALS_IGNORE_CASE = new MethodMatcher("String equalsIgnoreCase(java.lang.String)");
+
+    MethodMatcher EQUALS = new MethodMatcher("java.lang.String equals(java.lang.Object)");
+    MethodMatcher EQUALS_IGNORE_CASE = new MethodMatcher("java.lang.String equalsIgnoreCase(java.lang.String)");
+    MethodMatcher COMPARE_TO = new MethodMatcher("java.lang.String compareTo(java.lang.String)");
+    MethodMatcher COMPARE_TO_IGNORE_CASE = new MethodMatcher("java.lang.String compareToIgnoreCase(java.lang.String)");
+    MethodMatcher CONTENT_EQUALS = new MethodMatcher("java.lang.String contentEquals(java.lang.CharSequence)");
 
     EqualsAvoidsNullStyle style;
 
     @Override
     public J visitMethodInvocation(J.MethodInvocation method, P p) {
-        J j = super.visitMethodInvocation(method, p);
-        if (!(j instanceof J.MethodInvocation)) {
-            return j;
-        }
-        J.MethodInvocation m = (J.MethodInvocation) j;
-        if (m.getSelect() == null) {
-            return m;
+        return getSuperIfSelectNull((J.MethodInvocation) super.visitMethodInvocation(method, p));
+    }
+
+    private J getSuperIfSelectNull(final J.MethodInvocation methodInvocation) {
+        return isNull(methodInvocation.getSelect()) ?
+                methodInvocation :
+                !(methodInvocation.getSelect() instanceof J.Literal)
+                        && methodInvocation.getArguments().get(0) instanceof J.Literal
+                        && isStringComparisonMethod(methodInvocation)
+                        ? handleNullSafetyCheck(methodInvocation, getCursor().getParentTreeCursor().getValue())
```

Comment:
I am unsure of the original intent behind the current code, so please review and revise the methods or provide suggestions for improving the structure. I believe the current code is somewhat acceptable, but dividing each function into separate units would better adhere to the Single Responsibility Principle. This approach will focus on reducing redundancy, ensuring each function has a distinct and well-defined purpose, and avoiding 'feature envy' by keeping responsibilities isolated.